src/test/java/com/uber/jenkins/phabricator/conduit/DifferentialClientTest.java

```diff
@@ -0,0 +1,124 @@
+import net.sf.json.JSONObject;
+import net.sf.json.groovy.JsonSlurper;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import static junit.framework.TestCase.assertEquals;
+import static org.mockito.Matchers.*;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.spy;
+
+public class DifferentialClientTest {
+    @Rule
+    public JenkinsRule j = new JenkinsRule();
+
+    private final String DUMMY_DIFF_ID = "123";
+    private final String DUMMY_CONDUIT_TOKEN = "notarealtoken";
+    private final String DUMMY_ARC_PATH = "echo";
+
+    private DifferentialClient client;
+
+    @Before
+    public void setUp() throws Exception {
+        client = createClient();
+    }
+
+    @Test
+    public void testPostComment() throws Exception {
+        JSONObject sentinel = new JSONObject();
+        sentinel.put("hi", "there");
+
+        mockConduitResponse(client, sentinel);
+
+        JSONObject response = client.postComment("hello", true, "none");
+        assertEquals(sentinel, response);
+    }
+
+    @Test
+    public void testPostCommentSingleArgument() throws Exception {
+        JSONObject sentinel = new JSONObject();
+        sentinel.put("something", "here");
+
+        mockConduitResponse(client, sentinel);
+
+        JSONObject response = client.postComment("hello");
+        assertEquals(response, sentinel);
+    }
+
+    @Test(expected = ArcanistUsageException.class)
+    public void testFetchDiffWithEmptyResponse() throws Exception {
+        JSONObject empty = new JSONObject();
+        mockConduitResponse(client, empty);
+
+        client.fetchDiff();
+    }
+
+    @Test(expected = ArcanistUsageException.class)
+    public void testFetchDiffWithNoDiff() throws Exception {
+        JSONObject noDiff = new JSONObject();
+        noDiff.put("response", null);
+        mockConduitResponse(client, noDiff);
+
+        client.fetchDiff();
+    }
+
+    @Test
+    public void testFetchDiffWithValidResponse() throws Exception {
+        JSONObject realResponse = getValidFetchDiffResponse();
+        mockConduitResponse(client, realResponse);
+
+        JSONObject response = client.fetchDiff();
+        assertEquals("world", response.get("hello"));
+    }
+
+    private void mockConduitResponse(DifferentialClient client, JSONObject response) throws InterruptedException, ArcanistUsageException, IOException {
+        doReturn(response).when(client).callConduit(
+                anyString(),
+                anyMap()
+        );
+    }
+
+    private DifferentialClient createClient() throws Exception {
+        return spy(new DifferentialClient(
+                DUMMY_DIFF_ID,
+                TestUtils.createLauncherFactory(j),
+                DUMMY_CONDUIT_TOKEN,
+                DUMMY_ARC_PATH
+        ));
+    }
+
+    private JSONObject getValidFetchDiffResponse() throws IOException {
```

Comment:
Let's make a generic method in TestUtils that returns a JSONObject from a resource. Make that depend on another method that just returns a raw string from a resource.